src/parser/core/Combatant.ts

```diff
@@ -167,11 +167,7 @@ class Combatant extends Entity {
     if (spellObj.id) {
       spellId = spellObj.id;
     }
-    return Boolean(
-      Object.keys(this._talentsByRow).find(
-        (row: string) => this._talentsByRow[Number(row)] === spellId,
-      ),
-    );
```

Comment:
if the performance of this is a problem, we should store both `_talentSet: Set<number>` and `_talentsByRow: Record<number, number>` and use them as appropriate---trading memory for runtime.

if we don't want to store it twice, then simplifying this to `return Object.values(this._talentsByRow).find(id => id === spellId) !== undefined;` is likely faster and also simpler code.